File: SDKs/shutterstock.com/1.1.5/typescript/src/sdk/models/shared/audio.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Album } from "./album";
import { Artist } from "./artist";
import { AudioAssets } from "./audioassets";
import { Contributor } from "./contributor";
import { ModelRelease } from "./modelrelease";



// Audio
/** 
 * Audio metadata
**/
export class Audio extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=added_date" })
  addedDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=affiliate_url" })
  affiliateUrl?: string;

  @SpeakeasyMetadata({ data: "json, name=album" })
  album?: Album;

  @SpeakeasyMetadata({ data: "json, name=artists", elemType: Artist })
  artists?: Artist[];

  @SpeakeasyMetadata({ data: "json, name=assets" })
  assets?: AudioAssets;

  @SpeakeasyMetadata({ data: "json, name=bpm" })
  bpm?: number;

  @SpeakeasyMetadata({ data: "json, name=contributor" })
  contributor: Contributor;

  @SpeakeasyMetadata({ data: "json, name=deleted_time" })
  deletedTime?: Date;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description?: string;

  @SpeakeasyMetadata({ data: "json, name=duration" })
  duration?: number;

  @SpeakeasyMetadata({ data: "json, name=genres" })
  genres?: string[];

  @SpeakeasyMetadata({ data: "json, name=id" })
  id: string;

  @SpeakeasyMetadata({ data: "json, name=instruments" })
  instruments?: string[];

  @SpeakeasyMetadata({ data: "json, name=is_adult" })
  isAdult?: boolean;

  @SpeakeasyMetadata({ data: "json, name=is_instrumental" })
  isInstrumental?: boolean;

  @SpeakeasyMetadata({ data: "json, name=isrc" })
  isrc?: string;

  @SpeakeasyMetadata({ data: "json, name=keywords" })
  keywords?: string[];

  @SpeakeasyMetadata({ data: "json, name=language" })
  language?: string;

  @SpeakeasyMetadata({ data: "json, name=lyrics" })
  lyrics?: string;

  @SpeakeasyMetadata({ data: "json, name=media_type" })
  mediaType: string;

  @SpeakeasyMetadata({ data: "json, name=model_releases", elemType: ModelRelease })
  modelReleases?: ModelRelease[];

  @SpeakeasyMetadata({ data: "json, name=moods" })
  moods?: string[];

  @SpeakeasyMetadata({ data: "json, name=published_time" })
  publishedTime?: Date;

  @SpeakeasyMetadata({ data: "json, name=recording_version" })
  recordingVersion?: string;

  @SpeakeasyMetadata({ data: "json, name=releases" })
  releases?: string[];

  @SpeakeasyMetadata({ data: "json, name=similar_artists", elemType: Artist })
  similarArtists?: Artist[];

  @SpeakeasyMetadata({ data: "json, name=submitted_time" })
  submittedTime?: Date;

  @SpeakeasyMetadata({ data: "json, name=title" })
  title?: string;

  @SpeakeasyMetadata({ data: "json, name=updated_time" })
  updatedTime?: Date;

  @SpeakeasyMetadata({ data: "json, name=url" })
  url?: string;

  @SpeakeasyMetadata({ data: "json, name=vocal_description" })
  vocalDescription?: string;
}

File: SDKs/shutterstock.com/1.1.5/python/src/sdk/models/shared/audio.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import album as shared_album
from ..shared import artist as shared_artist
from ..shared import audioassets as shared_audioassets
from ..shared import contributor as shared_contributor
from ..shared import modelrelease as shared_modelrelease


@dataclass_json
@dataclasses.dataclass
class Audio:
    r"""Audio
    Audio metadata
    """
    
    contributor: shared_contributor.Contributor = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('contributor') }})
    id: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    media_type: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('media_type') }})
    added_date: Optional[date] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('added_date'), 'encoder': utils.dateisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    affiliate_url: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('affiliate_url') }})
    album: Optional[shared_album.Album] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('album') }})
    artists: Optional[list[shared_artist.Artist]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('artists') }})
    assets: Optional[shared_audioassets.AudioAssets] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('assets') }})
    bpm: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('bpm') }})
    deleted_time: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('deleted_time'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    description: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    duration: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('duration') }})
    genres: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('genres') }})
    instruments: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('instruments') }})
    is_adult: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_adult') }})
    is_instrumental: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_instrumental') }})
    isrc: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('isrc') }})
    keywords: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('keywords') }})
    language: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('language') }})
    lyrics: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('lyrics') }})
    model_releases: Optional[list[shared_modelrelease.ModelRelease]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('model_releases') }})
    moods: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('moods') }})
    published_time: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('published_time'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    recording_version: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('recording_version') }})
    releases: Optional[list[str]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('releases') }})
    similar_artists: Optional[list[shared_artist.Artist]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('similar_artists') }})
    submitted_time: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('submitted_time'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    title: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('title') }})
    updated_time: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('updated_time'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    url: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('url') }})
    vocal_description: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('vocal_description') }})
    
